### ISSUE TITLE
Aliasing a concrete type causes a RecursionError for identity map

### ISSUE THREAD

**Issue by mrogaski** [meadsteve/lagom#159]
It appears that the change to how the `Alias.skip_definitions` flag is used in #150 introduced a regression.  If an alias maps a type to itself, any resolution to the alias will raise a RecursionError.

```
Python 3.9.4 (tags/v3.9.4:1f2e308, Apr  6 2021, 13:40:21) [MSC v.1928 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import abc
>>> import sys
>>> import lagom
>>> sys.tracebacklimit = 5
>>>
>>> class AbstractFoo(abc.ABC):
...   pass
...
>>> class Foo(AbstractFoo):
...   pass
...
>>> container = lagom.Container()
>>>
>>> container[AbstractFoo] = Foo
>>> container[AbstractFoo]
<__main__.Foo object at 0x000001BF1CA4D700>
>>> container[Foo]
<__main__.Foo object at 0x000001BF1CA4D790>
>>>
>>> container[Foo] = Foo
>>> container[AbstractFoo]
Traceback (most recent call last):
  File "X:\lagom\.venv\lib\site-packages\lagom\definitions.py", line 68, in get_instance
    return container.resolve(
  File "X:\lagom\.venv\lib\site-packages\lagom\container.py", line 237, in resolve
    return definition.get_instance(self)
  File "X:\lagom\.venv\lib\site-packages\lagom\definitions.py", line 68, in get_instance
    return container.resolve(
  File "X:\lagom\.venv\lib\site-packages\lagom\container.py", line 235, in resolve
    definition = self.get_definition(dep_type)
  File "X:\lagom\.venv\lib\site-packages\lagom\container.py", line 352, in get_definition
    definition = self._registered_types.get(dep_type, Unset)
RecursionError: maximum recursion depth exceeded while calling a Python object
>>> container[Foo]
Traceback (most recent call last):
  File "X:\lagom\.venv\lib\site-packages\lagom\definitions.py", line 68, in get_instance
    return container.resolve(
  File "X:\lagom\.venv\lib\site-packages\lagom\container.py", line 237, in resolve
    return definition.get_instance(self)
  File "X:\lagom\.venv\lib\site-packages\lagom\definitions.py", line 68, in get_instance
    return container.resolve(
  File "X:\lagom\.venv\lib\site-packages\lagom\container.py", line 235, in resolve
    definition = self.get_definition(dep_type)
  File "X:\lagom\.venv\lib\site-packages\lagom\container.py", line 352, in get_definition
    definition = self._registered_types.get(dep_type, Unset)
RecursionError: maximum recursion depth exceeded while calling a Python object
>>>

```

This is Python 3.9.4 and Lagom 1.3.0.

Ultimately, this isn't a huge issue.  But it seems that a guard condition is merited so the identity map can be allowed.

**Comment by mrogaski** [comment]
Also, thank you for this wonderful package!

**Comment by meadsteve** [comment]
Thanks for the detailed description @mrogaski I'll take a look at this.

**Comment by meadsteve** [comment]
I've just released 1.3.1 that should fix the regression. Let me know if it doesn't. Thanks again for the detailed report.
